### ISSUE TITLE
train_val.prototxt file for fine-tuning

### ISSUE THREAD

**Issue by michaelholm-ce** [yahoo/open_nsfw#8]
Hello.  Thank you for sharing this project.  The readme suggests fine-tuning for improved performance, and I'm writing to request direction on how to convert the deploy.prototxt file into a train_val.prototxt file for fine-tuning.  Or, more simply, to request a sample train_val.prototxt file.

Thank you.

**Comment by jay-mahadeokar** [comment]
@michaelholm-ce the train and test prototxts were generated using [pynetbuilder](https://github.com/jay-mahadeokar/pynetbuilder).  Example the thin resnet prototxts for training on imagenet can be found [here](https://github.com/jay-mahadeokar/pynetbuilder/tree/master/models/imagenet/resnet_50_1by2)    You can modify the [FC1000 layer](https://github.com/jay-mahadeokar/pynetbuilder/blob/master/models/imagenet/resnet_50_1by2/train.prototxt#L3467-L3489) and replace it with [fc_nsfw](https://github.com/yahoo/open_nsfw/blob/master/nsfw_model/deploy.prototxt#L3457-L3481) layer. Rest I believe everything should remain same.

**Comment by michaelholm-ce** [comment]
Will you please share hyperparameters that are good candidates for training (e.g. solver.prototxt)?  I have tried many varieties through Nvidia Digits, so far to no avail with this model architecture.

**Comment by ProGamerGov** [comment]
I took a stab at creating the train_val.prototxt as per @jay-mahadeokar 's instructions: https://gist.github.com/ProGamerGov/2dedda3ad769fbe322e9b5db63db7763

Curious to see how it work with https://github.com/jcjohnson/fast-neural-style.

**Comment by michaelholm-ce** [comment]
I have implemented a similar train_val.prototxt file, but (via Nvidia DIGITS), the model will not learn (on a data set that Alexnet and Googlenet easily learn on), in spite of lowering learning rates and increasing batch sizes.  Any suggestions are appreciated.  Dataset is 1 million images for each of two classes, and I am using 4 TitanX GPUs.

**Comment by jay-mahadeokar** [comment]
I would suggest experimenting with hyper parameters for best results. Following could be a good starting point:

base_lr: 0.001
lr_policy: "step"
stepsize: [5 epochs?, could be again tuned]  
gamma: 0.5

**Comment by lixiaodongcq** [comment]
Hi, if i get the train_val.prototxt file. How can I label the training data?
